lib/record_store/provider/ns1/client.rb

```diff
@@ -1,46 +1,58 @@
+require 'net/http'
 require 'ns1'
 
 module RecordStore
   class Provider::NS1 < Provider
-    class Error < StandardError; end
-
     class Client < ::NS1::Client
       def initialize(api_key:)
         super(api_key)
       end
 
       def zones
-        super
+        zones = super
+        raise_error(zones) if zones.is_a?(NS1::Response::Error)
```

Comment:
May I suggest updating your `raise_error` into `raise_if_error!` and doing `return unless param.is_a?(NS1::Response::Error)` as first line of that method ? That would allow to avoid all the repeated ifs here.